protocol/eth-shared-lockbox.md

```diff
@@ -0,0 +1,192 @@
+- **Replacing chain servicers** for existing chains if they fail to satisfy technical requirements.
+
+### Shared Bridging and SuperchainWETH usage
+
+In any OP Chain that joins the cluster, it is assumed that `SuperchainWETH` has been deployed in advance prior to being added to the interoperable graph. As a result, the equivalence between ETH deposits and withdrawal history and the actual ETH supply will vary from the outset. In a world with freedom of movement, all real ETH liquidity is theoretically shared across the entire cluster eventually, regardless of how deposits are handled.
+
+# Solution
+
+The existing problem and considerations motivate us to propose an L1 shared liquidity design through the introduction of a new `SharedLockbox` on L1, which serves as a singleton contract for ETH, given a defined set of interoperable chains. To ensure consistency, the `SuperchainConfig` is extended to act as the *dependency set manager*, controlling the op-governed dependency set that the lockbox uses as a source of truth. New ETH deposits will be directed to the lockbox, with the same process applied to ETH withdrawals.
+
+### Spec changes
+The core proposed changes are as follows:
+- **Modify the `SuperchainConfig` contract**: Extend this contract to manage the dependency set of each chain, taking ownership of the `dependencyManager` role for each `SystemConfig`.
+- **Introduce the `SharedLockbox` contract**: This contract acts as an escrow for ETH, receiving deposits and allowing withdrawals from approved `OptimismPortal` contracts. The `SuperchainConfig` contract serves as the source of truth of the lockbox.
+- **Modify the `OptimismPortal`**: To forward ETH into the `SharedLockbox` when `depositTransaction` is called, with the inverse process applying when `finalizeWithdrawal` is called.
+
+### Managing op-governed dependency set through `SuperchainConfig`
+
+The `SuperchainConfig` contract will serve as the single point for managing the dependency set of a cluster and will be managed by the admin as occurs in other L1 OP contracts. This contract assumes the role of `dependencyManager` for every `SystemConfig` contract involved. Since the dependency graph follows a simple dependency, it only stores a mapping (or array) of chains added, e.g. given a `chainId`.
+
+```solidity
+// Mapping from chainId to SystemConfig address
+mapping(uint256 _chainId => ISystemConfig) public systemConfigs;
+
+// Current dependency set list
+EnumerableSet.UintSet private _dependencySet;
+```
+
+Adding a new chain can be done by introducing a new function called `addChain`. The process would look as follows:
+
+1. The `updater` calls the function with `chainId` and `SystemConfig`.
+2. An external call is made to each `SystemConfig` from already added chains with the new `chainId`.
+3. The new `SystemConfig` is called to add all previously existing `chainId` from `SuperchainConfig` to itself.
```

Comment:
Would we need to enforce via social that the dependency set is empty before `addChain` is called? Perhaps we keep dependency set size in the `SystemConfig` for this reason, so you can `require(config.depSetSize() == 0)`

Reply:
We would need to track the dependency set size for each chain in `SystemConfig`, updating it by adding or subtracting in the `addDependency` and `removeDependency` functions. The size check will apply only to the new chain being added at `addChain` in the `SuperchainConfig`. Does this approach make sense?

Reply:
Yes this makes sense, its double storage on L1 since the set is stored in both sys config + superchain config under this model. Not the end of the world but we generally need a way to make sure that the dependency set size is 0 before the chain joins